cloud.foundry.cli/src/main/java/cloud/foundry/cli/operations/ApplicationOperations.java

```diff
@@ -19,20 +34,199 @@ public ApplicationOperations(DefaultCloudFoundryOperations cloudFoundryOperation
+                .onErrorContinue(throwable -> throwable instanceof IllegalArgumentException
+                                //Fatal errors, exclude them.
+                                && !throwable.getMessage().contains("Application")
+                                && !throwable.getMessage().contains("Stack"),
+                        //TODO log warn instead
+                        (throwable, o) -> System.out.println("Warning: " + throwable.getMessage()))
+                //Error when staging or starting. So don't throw error, only log error.
+                .onErrorContinue(throwable -> throwable instanceof IllegalStateException,
+                        //TODO log warn instead
+                        (throwable, o) -> System.out.println("Warning: " + throwable.getMessage()))
+                .block();
+    }
+
+    private ApplicationManifest buildApplicationManifest(ApplicationBean bean) {
+        ApplicationManifest.Builder builder = ApplicationManifest.builder();
+
+        builder
+            .name(bean.getName())
+            .path(bean.getPath() == null ? null : Paths.get(bean.getPath()));
+
+        if (bean.getManifest() != null) {
+            builder.buildpack(bean.getManifest().getBuildpack())
+                .command(bean.getManifest().getCommand())
+                .disk(bean.getManifest().getDisk())
+                .docker(Docker.builder()
+                        .image(bean.getManifest().getDockerImage())
+                        .username(bean.getManifest().getDockerUsername())
+                        .password(getDockerPassword(bean))
+                        .build())
+                .healthCheckHttpEndpoint(bean.getManifest().getHealthCheckHttpEndpoint())
+                .healthCheckType(bean.getManifest().getHealthCheckType())
+                .instances(bean.getManifest().getInstances())
+                .memory(bean.getManifest().getMemory())
+                .noRoute(bean.getManifest().getNoRoute())
+                .routePath(bean.getManifest().getRoutePath())
+                .randomRoute(bean.getManifest().getRandomRoute())
+                .routes(getAppRoutes(bean.getManifest().getRoutes()))
+                .stack(bean.getManifest().getStack())
+                .timeout(bean.getManifest().getTimeout())
+                .putAllEnvironmentVariables(Optional.ofNullable(bean.getManifest().getEnvironmentVariables())
+                        .orElse(Collections.emptyMap()))
+                .services(bean.getManifest().getServices());
+        }
+
+        return builder.build();
+    }
+
+    private String getDockerPassword(ApplicationBean bean) {
+       if (bean.getManifest().getDockerImage() == null  || bean.getManifest().getDockerUsername() == null) {
+           return null;
+       }
+
+       //TODO: Maybe outsource retrieving env variables to a dedicated class in a future feature.
+       return System.getenv("CF_DOCKER_PASSWORD");
```

Comment:
Possible Nullpointer/Security Excpetion here. Maybe we should throw it here.

Reply:
added exception handling